plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py

```diff
@@ -309,68 +310,55 @@ def remove_unpinned_files(self) -> None:
     def download(
         self,
         hash_id: str,
-        target_dir: str,
+        target_dir: Union[str, Path],
         fix_path: bool = True,
         attempts: int = 5,
     ) -> str:
         """
         Download dir by its hash.
 
-        :param hash_id: str. hash of file to download
-        :param target_dir: str. directory to place downloaded
+        :param hash_id: str. hash of file or package to download
+        :param target_dir: Union[str, Path]. directory to place downloaded
         :param fix_path: bool. default True. on download don't wrap result in to hash_id directory.
         :param attempts: int. default 5. How often to attempt the download.
         :return: downloaded path
         """
 
-        if not os.path.exists(target_dir):  # pragma: nocover
-            os.makedirs(target_dir, exist_ok=True)
+        def move_to_target_dir(download_path) -> str:
+            """Move downloaded content to target directory"""
+
+            if download_path.is_file():
+                shutil.copy(download_path, target_dir)
+                return str(target_dir)
+
+            # else it is a directory containing a single package path
+            package_path = download_path
+            if fix_path:
+                # assumption is it contains one nested directory: the package
+                paths = list(download_path.glob("*"))
+                assumption_is_valid = len(paths) == 1 and paths[0].is_dir()
+                if not assumption_is_valid:  # pragma: no cover
+                    error_msg = f"Expected a single directory, found: {paths}"
+                    raise DownloadError(error_msg)
+                package_path = paths.pop()
 
-        if os.path.exists(os.path.join(target_dir, hash_id)):  # pragma: nocover
-            raise DownloadError(f"{hash_id} was already downloaded to {target_dir}")
```

Comment:
no longer raising here. `ipfshttpclient.Client.get` also overwrites if the path already exists: if your local is different, your local is wrong. 

Furthermore, if we `fix_path` and effectively remove the content up one directory, this check becomes useless.